Merge `gh-pages` into `master` to create unified repo

mavaddat opened the pull request:
I have merged the specific GitHub pages `gh-pages` branch back into the `master` so that the README.md files shown in the master are consistent with what is in the GitHub page for the repo. Also, several linguistic changes have been proposed.

ehmatthes committed:
Create gh-pages branch via GitHub

ehmatthes committed:
I believe gh pages is set up properly locally, and to push live.

ehmatthes committed:
Updates page renders using default layout.

ehmatthes committed:
Simplified header for subpages.

ehmatthes committed:
Project subtitle links to nostarch book site and nostarch home page.

ehmatthes committed:
Home page links to book and nostarch pages.

ehmatthes committed:
Subpages link back to main page.

ehmatthes committed:
Set baseurl to use stylesheets correctly for local dev and deployed pages.

ehmatthes committed:
Stylesheets and chapter 1 readme should work locally and live.

ehmatthes committed:
Copied all md files to pages branch.

ehmatthes committed:
Replaced all .md with .html.

ehmatthes committed:
All links seem to work.

ehmatthes committed:
Updated copy on home page.

ehmatthes committed:
Added contact information to updates page.

ehmatthes committed:
Added section about cheat sheets on index page.

ehmatthes committed:
Added image of cover to index and cheatsheets.

ehmatthes committed:
Fixed issue with required space after pound symbols for headers.

ehmatthes committed:
Added missing image files, and fixed links on updates page.

ehmatthes committed:
Updated cheat sheets page to reference if_while sheet.

ehmatthes committed:
Added google analytics.

ehmatthes committed:
Added files_exceptions sheet, updated copy.

ehmatthes committed:
Corrected link to files_exceptions pdf.

ehmatthes committed:
Added testing cheatsheet.

ehmatthes committed:
Added pygame cheatsheet.

ehmatthes committed:
Added matplotlib cheatsheet.

ehmatthes committed:
Added Pygal cheatsheet.

ehmatthes committed:
Added Django cheatsheet.

ehmatthes committed:
Updated links to point to release assets.

ehmatthes committed:
Pygal reverted default settings so most of the book's code works again!

ehmatthes committed:
Formatting code.

ehmatthes committed:
Updated installation instructions for matplotlib.

ehmatthes committed:
Started chapter 2 solutions.

ehmatthes committed:
Fixed link?

ehmatthes committed:
Added 2-5.

ehmatthes committed:
Fixed indent level for code.

ehmatthes committed:
Finished chapter 2 exercises.

ehmatthes committed:
Links to each problem at top of page.

ehmatthes committed:
Link back to solutions page.

ehmatthes committed:
Links to top after each solution.

ehmatthes committed:
Started chapter 3 solutions.

ehmatthes committed:
Linked from home page to solutions.

ehmatthes committed:
Through exercise 3-5.

ehmatthes committed:
Comment on first line affects code style?

ehmatthes committed:
No, maybe need an extra blank line?

ehmatthes committed:
Using fenced code block.

ehmatthes committed:
Python syntax highlighting.

ehmatthes committed:
Exercise 3-6.

ehmatthes committed:
Exercise 3-7.

ehmatthes committed:
Finished 3-8.

ehmatthes committed:
Set highlighting to rouge?

ehmatthes committed:
Trying highlight tag.

ehmatthes committed:
Added syntax.css stylesheet.

ehmatthes committed:
Trying kramdown spec in config.

ehmatthes committed:
Simplified config.

ehmatthes committed:
Added python highlighting to all code in solutions.

ehmatthes committed:
Converted remaining blocks in ch 3 to fenced blocks with python highlighting.

ehmatthes committed:
Through exercise 4-5.

ehmatthes committed:
Show output?

ehmatthes committed:
Need blank line before backticks?

ehmatthes committed:
Added output for chapter 4 programs.

ehmatthes committed:
Added output to chapter 2.

ehmatthes committed:
Added output to chapter 3.

ehmatthes committed:
Through 4-9.

ehmatthes committed:
Finished chapter 4 exercises.

ehmatthes committed:
Started chapter 5.

ehmatthes committed:
Added link to chapter 5 solutions.

ehmatthes committed:
Through exercise 5-5.

ehmatthes committed:
Through 5-7.

ehmatthes committed:
Through 5-11.

ehmatthes committed:
Added copy to solutions home page.

ehmatthes committed:
Through 6-3.

ehmatthes committed:
Through 6-5.

ehmatthes committed:
Through 6-6.

ehmatthes committed:
Through 6-8.

ehmatthes committed:
Finished chapter 6.

ehmatthes committed:
Fixed backtick issue.

ehmatthes committed:
Started chapter 7.

ehmatthes committed:
Bolded user input.

ehmatthes committed:
7-2, bolding input attempt.

ehmatthes committed:
Fixed bolding of input.

ehmatthes committed:
Through 7-3.

ehmatthes committed:
Through 7-5.

ehmatthes committed:
Finished 7-8.

ehmatthes committed:
Finished chapter 7.

ehmatthes committed:
Bolded more input.

ehmatthes committed:
Through 8-2.

ehmatthes committed:
8-3.

ehmatthes committed:
Through 8-5.

ehmatthes committed:
Through 8-8.

ehmatthes committed:
Trying pre over backticks.

ehmatthes committed:
pre formatting is better than backticks for output with syntax.

ehmatthes committed:
Through 8-11.

ehmatthes committed:
Through 8-14.

ehmatthes committed:
Finished chapter 8.

ehmatthes committed:
Fixed name of file.

ehmatthes committed:
Started chapter 9.

ehmatthes committed:
Better example restaurant.

ehmatthes committed:
Through 9-3.

ehmatthes committed:
More consistent example.

ehmatthes committed:
Fixed links back to solutions page, added 9-4.

ehmatthes committed:
Through 9.5.

ehmatthes committed:
Through 9-8.

ehmatthes committed:
Fixed missing backtick.

ehmatthes committed:
9-9.

ehmatthes committed:
9-10.

ehmatthes committed:
9-11.

ehmatthes committed:
9-12.

ehmatthes committed:
9-13.

ehmatthes committed:
Finished chapter 9.

ehmatthes committed:
Started chapter 10.

ehmatthes committed:
Forgot output.

ehmatthes committed:
Through 10-2.

ehmatthes committed:
Through 10-5.

ehmatthes committed:
Through 10-6.

ehmatthes committed:
Through 10-9.

ehmatthes committed:
Labeled exercise file.

ehmatthes committed:
Finished chapter 10.

ehmatthes committed:
Revisions to chapter 10 copy.

ehmatthes committed:
Final ch 10 revisions.

ehmatthes committed:
Started chapter 11.

ehmatthes committed:
Through 11-2.

ehmatthes committed:
Finished chapter 11.

ehmatthes committed:
Started chapter 15.

ehmatthes committed:
15-2.

ehmatthes committed:
Fixed link to image.

ehmatthes committed:
15-3.

ehmatthes committed:
15-5.

ehmatthes committed:
Forgot header styling.

ehmatthes committed:
Added image to 15-5.

ehmatthes committed:
15-6.

ehmatthes committed:
15-7.

ehmatthes committed:
15-8.

ehmatthes committed:
Finished chapter 15.

ehmatthes committed:
Started chapter 16.

ehmatthes committed:
Added chart.

ehmatthes committed:
16-3.

ehmatthes committed:
16-4.

ehmatthes committed:
16-6.

ehmatthes committed:
Added update to ch 17, for project with description None.

ehmatthes committed:
Copyediting.

ehmatthes committed:
Finished chapter 16.

ehmatthes committed:
Fixed link to 16-8.

ehmatthes committed:
Update for chapter 17 - move font size from config to style.

ehmatthes committed:
Added header for final version of python_repos.py.

ehmatthes committed:
Added direct links from headers on updates page to chapter pages.

ehmatthes committed:
Fixed link to chapter 15 updates.

ehmatthes committed:
17-1.

ehmatthes committed:
17-2.

ehmatthes committed:
Finished chapter 17.

ehmatthes committed:
Fixed link and output styling.

ehmatthes committed:
Completed posting solutions.

ehmatthes committed:
Added links for so and r/learnpython.

ehmatthes committed:
Revised copy.

ehmatthes committed:
Updates to chapter 20, for /tmp and WhiteNoise.

ehmatthes committed:
Slight copyedits.

ehmatthes committed:
More specific link, copyedits.

ehmatthes committed:
Clarified process of switching to WhiteNoise.

ehmatthes committed:
Installing Pygame, libavcode -> libavcodec.

ehmatthes committed:
Match comments in online updates to updated book code.

ehmatthes committed:
Fixed issue in solution to ex 9-8 Privileges.

ehmatthes committed:
Trying to fix anchor links.

ehmatthes committed:
Fixed chapters 2-4 anchor links for solutions.

ehmatthes committed:
Fixed remaining anchor link issues in solutions.

ehmatthes committed:
Update on how to access CSV weather data from Weather Underground.

ehmatthes committed:
Updated copy.

ehmatthes committed:
Updates for encoding when opening files in chapter 10.

ehmatthes committed:
Added page numbers for word_count.py.

ehmatthes committed:
Updates for Django 2.0.

ehmatthes committed:
Copy edit, and fix image link?

ehmatthes committed:
Fixed image link.

ehmatthes committed:
Link from chapter 18 to chapter 19 updates.

ehmatthes committed:
Fixed link to chapter 19.

ehmatthes committed:
Added missing comma.

ehmatthes committed:
Slight corrections when entering changes for next printing.

ehmatthes committed:
Added note for exercise 6-8.

ehmatthes committed:
Updated solution for 16-6, because some years in the json file are unquoted now.

Commit:
Updates for Django 2.1, for chapter 19.

Commit:
Fixed internal links.

Commit:
Clarified text on main updates page.

Commit:
Updated to match WhiteNoise 4 configuration settings.

Commit:
Smaller padding on home page.

Commit:
Added links to survey.

Commit:
Added survey link to all sub pages.

Commit:
Added link to resources for second edition on all pages.

Commit:
Updated some of my ruby and jekyll installation.

Commit:
Added link to mailing list signup page, and fixed capitalization of title.

Commit:
Added link to updated version of cheat sheets.

Commit:
Some changes after moving to new computer.

Commit:
Updated json and bundler.

Commit:
Updated ga code.

Commit:
Added description of finding_employement section in second edition resources.

mavaddat committed:
Update README.md

mavaddat committed:
Update README.md

mavaddat committed:
Update README.md

mavaddat committed:
Update README.md

mavaddat committed:
Update README.md

mavaddat committed:
Merge branch 'master' of https://github.com/mavaddat-javid-education/pcc into gh-pages

mavaddat committed:
Update README.md

mavaddat committed:
Update README.md

clearer language

mavaddat committed:
Update README.md

fix split verb

mavaddat committed:
Update updates.md

Make text consistent with main README.md without repeating exact language

ehmatthes commented:
I'm sorry it wasn't clear, but I don't accept pull requests on this repository. This is a public repo so that people can access all of the code for Python Crash Course. I will look at turning off pull requests so there's no confusion on anyone else's part.

This is also a rather large pull request to spring on a repository maintainer without any prior communication. If this is frustrating to you at all, I would suggest you communicate with a project's maintainer before putting together such a significant change.

ehmatthes commented:
For the record, I just looked and I don't think there's any way to turn off pull requests. It's really on contributors to communicate with maintainers about whether a pull request is likely to be accepted or not.

mavaddat commented:
> I'm sorry it wasn't clear, but I don't accept pull requests on this repository. This is a public repo so that people can access all of the code for Python Crash Course. I will look at turning off pull requests so there's no confusion on anyone else's part.

Understood.

> This is also a rather large pull request to spring on a repository maintainer without any prior communication. If this is frustrating to you at all, I would suggest you communicate with a project's maintainer before putting together such a significant change.

I essentially merged the changes of the `gh-pages` branch into the `master`. All other changes are just linguistic suggestions. In case it is useful, here is a transcript of my git logs:

```bash
git checkout gh-pages
```

<details>
  <summary>Output of above command</summary>

```
Switched to a new branch 'gh-pages'
Branch 'gh-pages' set up to track remote branch 'gh-pages' from 'origin'.
```

</details>

```bash
git pull origin master --allow-unrelated-histories
```
<details>
  <summary>Output of above command</summary>
  
```
From https://github.com/mavaddat-javid-education/pcc
 * branch            master     -> FETCH_HEAD
CONFLICT (add/add): Merge conflict in chapter_20/README.md
Auto-merging chapter_20/README.md
CONFLICT (add/add): Merge conflict in chapter_19/README.md
Auto-merging chapter_19/README.md
CONFLICT (add/add): Merge conflict in chapter_18/README.md
Auto-merging chapter_18/README.md
CONFLICT (add/add): Merge conflict in chapter_17/README.md
Auto-merging chapter_17/README.md
CONFLICT (add/add): Merge conflict in chapter_16/README.md
Auto-merging chapter_16/README.md
CONFLICT (add/add): Merge conflict in chapter_15/README.md
Auto-merging chapter_15/README.md
CONFLICT (add/add): Merge conflict in chapter_12/installing_pip.md
Auto-merging chapter_12/installing_pip.md
CONFLICT (add/add): Merge conflict in chapter_12/README.md
Auto-merging chapter_12/README.md
CONFLICT (add/add): Merge conflict in chapter_01/windows_setup.md
Auto-merging chapter_01/windows_setup.md
CONFLICT (add/add): Merge conflict in chapter_01/osx_setup.md
Auto-merging chapter_01/osx_setup.md
CONFLICT (add/add): Merge conflict in chapter_01/linux_setup.md
Auto-merging chapter_01/linux_setup.md
CONFLICT (add/add): Merge conflict in chapter_01/figures/crash_course01-03.png
Auto-merging chapter_01/figures/crash_course01-03.png
CONFLICT (add/add): Merge conflict in chapter_01/figures/crash_course01-02.png
Auto-merging chapter_01/figures/crash_course01-02.png
CONFLICT (add/add): Merge conflict in chapter_01/figures/crash_course01-01.png
Auto-merging chapter_01/figures/crash_course01-01.png
CONFLICT (add/add): Merge conflict in chapter_01/README.md
Auto-merging chapter_01/README.md
CONFLICT (add/add): Merge conflict in appendix_a/README.md
Auto-merging appendix_a/README.md
CONFLICT (add/add): Merge conflict in .gitignore
Auto-merging .gitignore
Automatic merge failed; fix conflicts and then commit the result.
```
</details>
   
```bash
git stage *
git commit all
```
<details>
  <summary>Output of above command</summary>
  
```
[gh-pages ae4072f] Merge branch 'master' of https://github.com/mavaddat-javid-education/pcc into
 gh-pages
```
</details>

```bash
git push
```
<details>
  <summary>Output of above command</summary>
  
```
Enumerating objects: 76, done.
Counting objects: 100% (76/76), done.
Delta compression using up to 4 threads
Compressing objects: 100% (25/25), done.
Writing objects: 100% (26/26), 6.36 KiB | 1.27 MiB/s, done.
Total 26 (delta 19), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (19/19), completed with 19 local objects.
To https://github.com/mavaddat-javid-education/pcc.git
   57753e9..ae4072f  gh-pages -> gh-pages
```
</details>

```bash
git checkout master
```

<details>
  <summary>Output of above command</summary>
  
```
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
```
</details>

```bash
git merge gh-pages
```
<details>
  <summary>Output of above command</summary>
  
```
Updating 7d67f02..ae4072f
Fast-forward
 .gitignore                                |   11 +
 Gemfile                                   |    3 +
 Gemfile.lock                              |  130 ++++
 _config.yml                               |    7 +
 _layouts/default.html                     |   49 ++
 _layouts/home_page.html                   |   54 ++
 appendix_a/README.md                      |   31 +
 chapter_01/README.md                      |   15 +-
 chapter_01/figures/crash_course01-01.png  |  Bin
 chapter_01/figures/crash_course01-02.png  |  Bin
 chapter_01/figures/crash_course01-03.png  |  Bin
 chapter_01/linux_setup.md                 |    7 +
 chapter_01/osx_setup.md                   |    7 +
 chapter_01/windows_setup.md               |    7 +
 chapter_10/README.md                      |   17 +
 chapter_12/README.md                      |   28 +-
 chapter_12/installing_pip.md              |    7 +
 chapter_15/README.md                      |   66 +-
 chapter_16/README.md                      |   43 ++
 chapter_17/README.md                      |  172 ++++-
 chapter_18/README.md                      |  163 ++++-
 chapter_18/default_homepage.png           |  Bin 0 -> 56575 bytes
 chapter_19/README.md                      |   79 +++
 chapter_20/README.md                      |  121 ++++
 cheatsheets/README.md                     |   43 ++
 images/cover.jpg                          |  Bin 0 -> 16690 bytes
 images/cubes_5.png                        |  Bin 0 -> 29259 bytes
 images/cubes_5000.png                     |  Bin 0 -> 42164 bytes
 images/cubes_5000_colormap.png            |  Bin 0 -> 46751 bytes
 images/death_valley_same_scale.png        |  Bin 0 -> 155303 bytes
 images/dice_visual_2d6_multiplication.png |  Bin 0 -> 52696 bytes
 images/dice_visual_2d8.png                |  Bin 0 -> 53633 bytes
 images/dice_visual_3d6.png                |  Bin 0 -> 53898 bytes
 images/global_gdp.png                     |  Bin 0 -> 166117 bytes
 images/hn_discussions.png                 |  Bin 0 -> 116668 bytes
 images/js_repos.png                       |  Bin 0 -> 66491 bytes
 images/rw_50000.png                       |  Bin 0 -> 128926 bytes
 images/rw_molecular.png                   |  Bin 0 -> 68448 bytes
 images/rw_molecular_zorder.png            |  Bin 0 -> 77102 bytes
 images/sitka_death_valley.png             |  Bin 0 -> 211805 bytes
 images/sitka_rainfall_2015.png            |  Bin 0 -> 117742 bytes
 images/sitka_rainfall_2015_cumulative.png |  Bin 0 -> 80413 bytes
 images/sitka_weather_same_scale.png       |  Bin 0 -> 134188 bytes
 index.html                                |   78 +++
 params.json                               |    1 +
 resources/global_gdp.json                 |    1 +
 resources/sitka_rainfall_2015.csv         |  366 ++++++++++
 solutions/README.md                       |   31 +
 solutions/chapter_10.md                   |  623 +++++++++++++++++
 solutions/chapter_11.md                   |  221 ++++++
 solutions/chapter_15.md                   |  462 +++++++++++++
 solutions/chapter_16.md                   |  350 ++++++++++
 solutions/chapter_17.md                   |  281 ++++++++
 solutions/chapter_2.md                    |  117 ++++
 solutions/chapter_3.md                    |  450 ++++++++++++
 solutions/chapter_4.md                    |  302 ++++++++
 solutions/chapter_5.md                    |  337 +++++++++
 solutions/chapter_6.md                    |  550 +++++++++++++++
 solutions/chapter_7.md                    |  318 +++++++++
 solutions/chapter_8.md                    |  589 ++++++++++++++++
 solutions/chapter_9.md                    | 1067 +++++++++++++++++++++++++++++
 stylesheets/github-light.css              |  116 ++++
 stylesheets/my_styles.css                 |   24 +
 stylesheets/normalize.css                 |  424 ++++++++++++
 stylesheets/stylesheet.css                |  248 +++++++
 stylesheets/syntax.css                    |   65 ++
 updates.md                                |   19 +
 67 files changed, 8095 insertions(+), 5 deletions(-)
 mode change 100644 => 100755 .gitignore
 create mode 100755 Gemfile
 create mode 100755 Gemfile.lock
 create mode 100755 _config.yml
 create mode 100755 _layouts/default.html
 create mode 100755 _layouts/home_page.html
 mode change 100644 => 100755 appendix_a/README.md
 mode change 100644 => 100755 chapter_01/README.md
 mode change 100644 => 100755 chapter_01/figures/crash_course01-01.png
 mode change 100644 => 100755 chapter_01/figures/crash_course01-02.png
 mode change 100644 => 100755 chapter_01/figures/crash_course01-03.png
 mode change 100644 => 100755 chapter_01/linux_setup.md
 mode change 100644 => 100755 chapter_01/osx_setup.md
 mode change 100644 => 100755 chapter_01/windows_setup.md
 create mode 100755 chapter_10/README.md
 mode change 100644 => 100755 chapter_12/README.md
 mode change 100644 => 100755 chapter_12/installing_pip.md
 mode change 100644 => 100755 chapter_15/README.md
 mode change 100644 => 100755 chapter_16/README.md
 mode change 100644 => 100755 chapter_17/README.md
 mode change 100644 => 100755 chapter_18/README.md
 create mode 100755 chapter_18/default_homepage.png
 mode change 100644 => 100755 chapter_19/README.md
 mode change 100644 => 100755 chapter_20/README.md
 create mode 100755 cheatsheets/README.md
 create mode 100755 images/cover.jpg
 create mode 100755 images/cubes_5.png
 create mode 100755 images/cubes_5000.png
 create mode 100755 images/cubes_5000_colormap.png
 create mode 100755 images/death_valley_same_scale.png
 create mode 100755 images/dice_visual_2d6_multiplication.png
 create mode 100755 images/dice_visual_2d8.png
 create mode 100755 images/dice_visual_3d6.png
 create mode 100755 images/global_gdp.png
 create mode 100755 images/hn_discussions.png
 create mode 100755 images/js_repos.png
 create mode 100755 images/rw_50000.png
 create mode 100755 images/rw_molecular.png
 create mode 100755 images/rw_molecular_zorder.png
 create mode 100755 images/sitka_death_valley.png
 create mode 100755 images/sitka_rainfall_2015.png
 create mode 100755 images/sitka_rainfall_2015_cumulative.png
 create mode 100755 images/sitka_weather_same_scale.png
 create mode 100755 index.html
 create mode 100755 params.json
 create mode 100755 resources/global_gdp.json
 create mode 100755 resources/sitka_rainfall_2015.csv
 create mode 100755 solutions/README.md
 create mode 100755 solutions/chapter_10.md
 create mode 100755 solutions/chapter_11.md
 create mode 100755 solutions/chapter_15.md
 create mode 100755 solutions/chapter_16.md
 create mode 100755 solutions/chapter_17.md
 create mode 100755 solutions/chapter_2.md
 create mode 100755 solutions/chapter_3.md
 create mode 100755 solutions/chapter_4.md
 create mode 100755 solutions/chapter_5.md
 create mode 100755 solutions/chapter_6.md
 create mode 100755 solutions/chapter_7.md
 create mode 100755 solutions/chapter_8.md
 create mode 100755 solutions/chapter_9.md
 create mode 100755 stylesheets/github-light.css
 create mode 100755 stylesheets/my_styles.css
 create mode 100755 stylesheets/normalize.css
 create mode 100755 stylesheets/stylesheet.css
 create mode 100755 stylesheets/syntax.css
 create mode 100755 updates.md
```
</details>

```bash
git push
```

<details>
  <summary>Output of above command</summary>
  
```
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
To https://github.com/mavaddat-javid-education/pcc.git
   7d67f02..ae4072f  master -> master
```
</details>
